### PR TITLE
Fixed issue #16599: Cannot update answers (checkUniqueness bug)

### PR DESCRIPTION
**Pull request by febsn**
Dev: This fixes an issue with saving Type O (and maybe other) questions.

Dev: Introduces int casting for aid, qid and scale_id to prevent false
Dev: positive in Answer::checkUniqueness because of type mismatch.

Fixed issue #16599: Cannot update Answers (Type O) – Answer::checkUniqueness bug
Fixed issue #16313: Issues saving/modifying answers: "Answer codes must be unique by question"

### CODE REVIEW

**Comment by olleharstedt** [comment]
Looks like this breaks dual-scale question type?

File: application/controllers/QuestionEditorController.php [hunk 1]
```diff
@@ -1262,7 +1262,9 @@ private function storeAnswerOptions(&$oQuestion, $dataSet, $isCopyProcess = fals
         $this->cleanAnsweroptions($oQuestion, $dataSet);
         foreach ($dataSet as $aAnswerOptions) {
             foreach ($aAnswerOptions as $iScaleId => $aAnswerOptionDataSet) {
-                $aAnswerOptionDataSet['sortorder'] = (int) $aAnswerOptionDataSet['sortorder'];
+                foreach (['sortorder', 'aid', 'qid', 'scale_id'] as $key) {
```
Comment (olleharstedt) [inline application/controllers/QuestionEditorController.php:1265]: Please check Travis and the dual-scale question type.

**Comment by cdorin93** [comment]
Hey @febsn  , have you tried the fix provided in : https://bugs.limesurvey.org/view.php?id=16313

**Comment by cdorin93** [comment]
Closed: we reverted to the old view in master. (release expected next week the latest)

**Pull request closed without merging**